src/ZprimeSelectionModule.cxx

```diff
@@ -141,7 +144,13 @@ bool ZprimeSelectionModule::process(Event & event){
   jetlepton_cleaner->process(event);
   jetER_smearer->process(event);
   jet_cleaner1->process(event); // jets w/ pt>25 GeV for lepton-2Dcut
-  bool pass_twodcut = twodcut_sel->passes(event); // lepton 2D-cut
+  //Trigger selection
+  bool ele_Trigger_selection = ele_Trigger->passes(event);
+  bool mu_Trigger_selection = mu_Trigger->passes(event);
+  if (!ele_Trigger_selection || !mu_Trigger_selection || (mu_Trigger_selection && ele_Trigger_selection)) return false;
```

Comment:
I didn't get your point (esp. the "loading" two trigger selections, because this is not what ever would have happened), but now I think I do get it. To be concrete, you could do this as a member:
```
std::unique_ptr<Selection> trigger;
```

and then in the initialization:
```
if(muon){ ...
   trigger = make_unique<TriggerSelection>("HLT_Mu40_eta2p1_PFJet200_PFJet50_v*");
} else if(elec){ ...
   trigger = make_unique<TriggerSelection>("HLT_Ele45_CaloIdVT_GsfTrkIdT_PFJet200_PFJet50_v*");
}
```

and in process:
```
if(!trigger->passes(event)) return false;
```

This would use only one variable for the trigger selection instead of the two in this change, and it would avoid handling any lengthy boolean expressions.

Is this what you meant?

In any case, I think it would make the code clearer.